File: backend/src/main/java/com/inventory/backend/repository/HallRequestRepository.java
package com.inventory.backend.repository;

import com.inventory.backend.entity.HallRequest;
import com.inventory.backend.entity.HallRequestStatus;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Collection;
import java.util.List;

public interface HallRequestRepository extends JpaRepository<HallRequest, Long> {
    List<HallRequest> findAllByOrderByRequestedAtDesc();

    List<HallRequest> findByRoomIdOrderByRequestedAtDesc(Long roomId);

    List<HallRequest> findByStatusOrderByRequestedAtDesc(HallRequestStatus status);

    List<HallRequest> findByStatusInOrderByRequestedAtDesc(Collection<HallRequestStatus> statuses);
}

File: backend/src/main/kotlin/com/inventory/backend/repository/HallRequestRepository.kt
package com.inventory.backend.repository

import com.inventory.backend.entity.HallRequest
import com.inventory.backend.entity.HallRequestStatus
import org.springframework.data.jpa.repository.JpaRepository

interface HallRequestRepository : JpaRepository<HallRequest, Long> {
    fun findAllByOrderByRequestedAtDesc(): List<HallRequest>
    fun findByRoomIdOrderByRequestedAtDesc(roomId: Long): List<HallRequest>
    fun findByStatusOrderByRequestedAtDesc(status: HallRequestStatus): List<HallRequest>
    fun findByStatusInOrderByRequestedAtDesc(statuses: Collection<HallRequestStatus>): List<HallRequest>
}
